### PR TITLE
Inform user if update is available. Includes instuctions on how to update. Added Git to dependencies.

### PR DESCRIPTION
**Pull request by duncanjake1**
Updated ani-cli to check if the local repo is up to date with the remote repo on script start up. Included a message to be echo'd when an update is available or the local branch has become diverged.

Here's a nifty SO that explains the git syntax - https://stackoverflow.com/a/3278427/16004619

Added git to required dependencies in README.md and dep_ch method.

### CODE REVIEW

**Comment by duncanjake1** [comment]
Created this ticket in reaction to issue #124, Having this feature might help to clear up some confusion in the future.

**Review by ppcpwn — APPROVED** [review]

**Review by Beyley — APPROVED** [review]
lgtm

File: ani-cli [hunk 1]
```diff
@@ -127,6 +127,19 @@ get_links () {
 	printf '%s' "$video_url" | sed -n -E "s/(.*)\.m3u8/\1.$video_quality.m3u8/p"
 }
 
+check_for_update () {
+	git fetch 
+	UPSTREAM=${1:-'@{u}'}
+	LOCAL=$(git rev-parse @)
+	REMOTE=$(git rev-parse "$UPSTREAM")
+
+	UPDATE_MESSAGE="UPDATE AVAILABLE\nPlease run the following commands in your ani-cli directory to update\n----------\ngit pull\nsudo make\n----------\n"
```
Comment (Dink4n) [inline ani-cli:136]: Is there a need for UPDATE_MESSAGE since we only use it once?

File: ani-cli [hunk 2]
```diff
@@ -127,6 +127,19 @@ get_links () {
...
+
+	if [ $LOCAL != $REMOTE ]; then
+    echo $UPDATE_MESSAGE
```
Comment (Dink4n) [inline ani-cli:139]: Can you fix the indentation here? Also, instead of `echo`, use `printf`.

File: ani-cli (hunk 1 above)
Comment (Dink4n) [inline ani-cli:136]: Is there a need for `UPDATE_MESSAGE` since we only use it once?

Comment (duncanjake1) [reply ani-cli]: I removed the variable and made the text inline with printf

File: ani-cli (hunk 2 above)
Comment (duncanjake1) [reply ani-cli]: done and done

**Comment by duncanjake1** [comment]
I noticed that when the ani-cli command was ran outside of the ani-cli directory, the git fetch command would also run in that directory. This would end up running git fetch on a git directory that you didn't intend to, or printing out error text. 

To solve this, I added the ability to write the present directory into a variable in the ani-cli file at the moment that 'sudo make' is ran. git fetch will then be ran on that specific directory, no matter where you are when you run it.

**Review by RafaelKuhn — CHANGES_REQUESTED** [review]
Hey, my man, you shouldn't be PRing this with a path to a plnace inside your computer
line 2: ```ORIGINAL_DIR=/home/duncan/Apps/ani-cli2-fork```
maybe consider inserting that path there dynamically... 
for an example when the user runs make install...

**Comment by duncanjake1** [comment]
@RafaelKuhn That's actually the exact functionality that's happening now! In the last commit, I removed the hard coded path and entered a placeholder value that will be overwritten when the user runs the make file.

**Pull request merged**